packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/views/tree/DomainExplorerRenderVariablesCustomizer.java

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.application.views.tree;
+
+import java.util.List;
+import java.util.Objects;
+
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.representations.IRepresentationDescription;
+import org.eclipse.sirius.components.representations.IRepresentationRenderVariableCustomizer;
+import org.eclipse.sirius.components.representations.VariableManager;
+import org.eclipse.sirius.web.application.UUIDParser;
+import org.eclipse.sirius.web.application.studio.services.representations.DomainViewTreeDescriptionProvider;
+import org.eclipse.sirius.web.application.views.explorer.services.ExplorerDescriptionProvider;
+import org.eclipse.sirius.web.domain.boundedcontexts.representationdata.RepresentationMetadata;
+import org.eclipse.sirius.web.domain.boundedcontexts.representationdata.services.api.IRepresentationMetadataSearchService;
+import org.springframework.data.jdbc.core.mapping.AggregateReference;
+import org.springframework.stereotype.Service;
+
+/**
+ * Provides additional render variables to be used by {@link DomainExplorerRepresentationDescriptionProvider} to reduce the amount of duplicate work (notably database access):
+ *
+ * <dl>
+ * <dt>{@code existingRepresentations}<dt>
+ * <dd>the list of all {@RepresentationMetadata representations} present in the current editing context.</dd>
+ * </dl>
+ *
+ * @author pcdavid
+ */
+@Service
+public class DomainExplorerRenderVariablesCustomizer implements IRepresentationRenderVariableCustomizer {
+
+    private final IRepresentationMetadataSearchService representationMetadataSearchService;
+
+    public DomainExplorerRenderVariablesCustomizer(IRepresentationMetadataSearchService representationMetadataSearchService) {
+        this.representationMetadataSearchService = Objects.requireNonNull(representationMetadataSearchService);
+    }
+
+    @Override
+    public VariableManager customize(IRepresentationDescription representationDescription, VariableManager variableManager) {
+        if (DomainViewTreeDescriptionProvider.DOMAIN_EXPLORER_DESCRIPTION_NAME.equals(representationDescription.getLabel())) {
```

Comment:
Not very satisfying/reliable, but I could find a proper way to determine the actual id except hard-coding the value I see (`siriusComponents://representationDescription?kind=treeDescription&sourceKind=view&sourceId=ce7cce0b-2928-35d1-92f8-33a1d58fa0b0&sourceElementId=8ecf5948-3afd-3380-b783-c733323595a2`) given all the layers and transformations involved.

Reply:
I don't think you could do better for the moment, I'll just check your PR with a quick manual test and merge it